espresso/environment/4_confirmation_integrity_with_reorgs_test.go

```diff
@@ -0,0 +1,217 @@
+	str = str + b.ParentHash.String()
+
+	h := sha256.New()
+	h.Write([]byte(str))
+	hash := h.Sum(nil)
+
+	res := hex.EncodeToString(hash)
+
+	return res
+}
+
+// This function computes a list where several batches are sent to unfinalized L1 blocks.
+// It works as follows:
+//  1. Pick the current L1 block number. Store it so that later we can reorg back to this point
+//  2. Start from the height of the first L2 block unsafe block.
+//  3. Pick the blocks from this height and the subsequent.
+//     This first block is initially unsafe, but we wait for it to become safe inside the L1 finality window and likewise for the others.
+//  4. Do this until the "finality" window closes i.e. before the number of new L1 blocks is bigger than L1FinalizedDistance.
+//  5. While doing this also send transactions to the sequencer to avoid dealing with empty blocks.
+//  6. Return the list of batch hashes, the L1 block number to reorg to and also the L2 block height determined in step 2.
+//     @param ctx,t,system standard parameters to execute a test
+//     @param l1Client L1 client used to fetch L1 block numbers
+//     @param l2Seq sequencer used to send transactions
+//     @param l2Verif OP node we monitor for unsafe blocks
+//     @return batches list of hashes of the batches
+//     @return l1HeightStart L1 block number collected in step 1.
+//     @return unsafeL2BlockNumber L2 block number collected in step 2.
+func collectBatchesPublishedOnUnfinalizedL1Blocks(ctx context.Context, t *testing.T, system *e2esys.System, l1Client *ethclient.Client, l2Seq *ethclient.Client, l2Verif *ethclient.Client) ([]string, uint64, uint64) {
+	var batches []string
+
+	l1Height, err := l1Client.BlockNumber(ctx)
+	require.NoError(t, err)
+	l1HeightStart := l1Height
+	log.Info("L1 height to reorg to", "height", l1HeightStart)
+	// Keep monitoring L2 blocks while L1 is producing unfinalized blocks
+	i := int64(0)
+	// Fetch height of the most recent block which is unsafe and which batch will be sent to L1 at a later stage
+	rollupClient := system.RollupClient(e2esys.RoleVerif)
+
+	status, err := rollupClient.SyncStatus(ctx)
+	require.NoError(t, err)
+	unsafeL2BlockNumber := status.SafeL2.Number + 1
+
+	nonce := uint64(0)
+	addressAlice := system.Cfg.Secrets.Addresses().Alice
+
+	require.NoError(t, err)
+	for (l1Height - l1HeightStart) < system.Cfg.L1FinalizedDistance {
+		height := uint64(i) + unsafeL2BlockNumber
+
+		//Send some transactions to fill the batches
+		receipt := helpers.SendL2TxWithID(t, system.Cfg.L2ChainIDBig(), l2Seq, system.Cfg.Secrets.Bob, func(opts *helpers.TxOpts) {
+			opts.Nonce = nonce
+			opts.ToAddr = &addressAlice
+			opts.Value = new(big.Int).SetUint64(1)
+		})
+		nonce++
+		log.Info("Receipt", "value", receipt)
+
+		l2Head, err := geth.WaitForBlockToBeSafe(new(big.Int).SetUint64(height), l2Verif, 10*time.Second)
+		require.NoError(t, err)
+		
+		if err != nil {
+			continue
+		} else { // Insert new batch in the list
```

Comment:
`else` after `continue` seems to suggest that either
* The `else` is unnecessary because everything after it is executed under the same condition, or
* The `continue` is incorrect because we still want to proceed to the `l1Height` setting after the `else` clause.

Reply:
Indeed the else block is redundant! It has been removed in 00bbb9f1d.